lib/tasks/analyse/links.rb

```diff
@@ -75,26 +76,31 @@ def write_taxon_worksheet(results, spreadsheet)
       worksheet: worksheet,
     )
 
-    results_by_taxon_base_path = group_results(results, :taxon_base_path)
-
     row_number = 1
-    results_by_taxon_base_path.each_pair do |taxon_base_path, base_path_results|
-      navigation_page_type = base_path_results[0][:navigation_page_type]
 
-      if navigation_page_type == 'accordion'
-        base_path_results += base_path_results.map do |result|
-          result.merge(section: 'all')
+    results_by_navigation_page_type = group_results(results, :navigation_page_type)
```

Comment:
If you felt like it, you could possibly refactor `group_results` to accept any number of arguments so you could write something like:

```
grouped_results = group_results(results, :navigation_page_type, :taxon_base_path)
```

...which would slightly reduce nesting. But depends on if you think it's worth it. (After all, we can't collapse `:section` like this, because of the logic we've got in the middle - unless you can think of somewhere else to put that?).